testgen/generators/collation_short.py

```diff
@@ -78,8 +92,128 @@ def insert_collation_header(self, test_objs):
                 "UCA conformance test. Compare the first data string with the second and with strength = identical level (using S3.10). If the second string is greater than the first string, then stop with an error."
             )
 
+    def reset_test_data(self, rules, locale, attributes, strength):
+        rules = []
+        locale = ""
+        strength = None
+
+    def parse_compare(self, line_index, lines):
+        # Handles lines in a compare region
+        # Test sections ("* compare") are terminated by
+        # definitions of new collators, changing attributes, or new test sections.
+        tests = []
+        line_in = lines[line_index]
+        if not self.compare_pattern.match(line_in):
+            return None, line_index
+
+        string1 = ''
+        line_index += 1
+        while line_index < len(lines):
+            line_in = lines[line_index]
+
+            if self.compare_pattern.match(line_in):
+                break
+            if self.locale_string.match(line_in):
+                break
+            if self.root_locale.match(line_in):
+                break
+            if self.attribute_test.match(line_in):
+                break
+            if self.reorder_test.match(line_in):
+                break
+            if self.rule_header_pattern.match(line_in):
+                break
+            if self.test_line.match(line_in):
+                break
```

Comment:
could you do?
```suggestion
            unexpected_line_patterns = [
                self.compare_pattern,
                self.local_string,
                ...
            ]
            if any([p.match(line_in) for p in unexpected_line_patters]):
                break
```

Reply:
Also: should we break out of the loop, or is it an error? Would an error signal an assumption about the test file that isn't true? And if the assumption about the test file isn't true, is that a problem that we need to balk at, or is it okay to keep going on?